cpp/src/arrow/compute/key_map.h

```diff
@@ -226,12 +228,12 @@ class ARROW_EXPORT SwissTable {
   // ---------------------------------------------------
   // * Empty bucket has value 0x80. Non-empty bucket has highest bit set to 0.
   //
-  uint8_t* blocks_;
+  std::shared_ptr<Buffer> blocks_;
 
   // Array of hashes of values inserted into slots.
   // Undefined if the corresponding slot is empty.
   // There is 64B padding at the end.
```

Comment:
Should here still mention padding here? Since Buffer already ensure padding?

Reply:
Buffer ensures that allocations are aligned (and I think, at least 64 bytes) but I don't know if it actually guarantees padding.  I think this buffer needs to go 64 bytes past the end regardless of the size which is a slightly different requirement.

Reply:
Got it, padding and aligned are different, I missed it previously. the patch LGTM